### PR TITLE
feature(new integration):shynet

### PR DESCRIPTION
**Pull request by aashishmalik**
## Description of the change

> Onboarding new destination Shynety

## Type of change
- [ ] New feature (non-breaking change that adds functionality)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/shynet/transform.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+// process page call
+function processPage(message, shynetServiceUrl) {
+  const requestJson = constructPayload(
+    message,
+    mappingConfig[ConfigCategory.PAGE.name]
+  );
+
+  // generating UUID
+  requestJson.idempotency = generateUUID();
```
Comment (ItsSudip) [inline v0/destinations/shynet/transform.js:37]: Can't we use `anonymousId` or `userId`?

File: v0/destinations/shynet/transform.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  generateUUID
+} = require("../../util");
+
+const { ConfigCategory, mappingConfig } = require("./config");
+
+// build final response
+function buildResponse(payload, endpoint) {
+  const response = defaultRequestConfig();
+  response.endpoint = endpoint;
+  response.body.JSON = removeUndefinedAndNullValues(payload);
```
Comment (ItsSudip) [inline v0/destinations/shynet/transform.js:20]: Where is the REST API method?

File: v0/destinations/shynet/transform.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  return {
+    ...response,
+    headers: {
+      "Content-Type": "application/json"
+    }
+  };
```
Comment (ItsSudip) [inline v0/destinations/shynet/transform.js:26]:
```suggestion
    response.headers = {
      "Content-Type": "application/json"
    };
    return response;
  };
```

File: v0/destinations/shynet/data/ShynetPageConfig.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    "destKey": "referrer",
+    "sourceKeys": "context.page.referrer",
+    "required": false
+  },
+  {
+    "destKey": "location",
+    "sourceKeys": "context.page.url",
+    "required": false
+  }
+]
```
Comment (ItsSudip) [inline v0/destinations/shynet/data/ShynetPageConfig.json:17]: Are none of the fields required?

**Review by ItsSudip — CHANGES_REQUESTED** [review]

File: v0/destinations/shynet/transform.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+      "Message Type is not present. Aborting message.",
+      400
+    );
+  }
+
+  const messageType = message.type.toLowerCase();
+
+  switch (messageType) {
+    case EventType.PAGE:
+      message.event = message.event || "Visited a Page";
```
[the remark this answers is not in the document]
Comment (utsabc) [reply v0/destinations/shynet/transform.js:56]: https://www.rudderstack.com/docs/event-spec/standard-events/page/

File: v0/destinations/shynet/transform.js (hunk 2 above)
Comment (aashishmalik) [reply v0/destinations/shynet/transform.js:20]: default is post method only

File: v0/destinations/shynet/transform.js (hunk 1 above)
Comment (utsabc) [reply v0/destinations/shynet/transform.js:37]: I believe this is a random Id they need for each request

File: v0/destinations/shynet/transform.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (aashishmalik) [reply v0/destinations/shynet/transform.js:56]: removed this property

File: v0/destinations/shynet/data/ShynetPageConfig.json (hunk 4 above)
Comment (aashishmalik) [reply v0/destinations/shynet/data/ShynetPageConfig.json]: nothing mentioned as per doc, made url as required to atleast track the hit on url

File: v0/destinations/shynet/transform.js (hunk 3 above)
Comment (aashishmalik) [reply v0/destinations/shynet/transform.js]: done, was doing same thing though.

File: v0/destinations/shynet/transform.js (hunk 1 above)
Comment (ItsSudip) [reply v0/destinations/shynet/transform.js]: Can't we use messageId?

File: v0/destinations/shynet/data/ShynetPageConfig.json [hunk 6]
```diff
@@ -0,0 +1,17 @@
+[
+  {
+    "destKey": "loadTime",
+    "sourceKeys": "context.page.loadTime",
```
Comment (utsabc) [inline v0/destinations/shynet/data/ShynetPageConfig.json:4]: We don't have `loadTime` in `context.page` 

```suggestion
    "sourceKeys": "properties.loadTime"
```

**Comment by utsabc** [comment]
1 small comment rest looks good

**Review by utsabc — APPROVED** [review]

**Review by ItsSudip — DISMISSED** [review]

**Review by utsabc — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Pull request merged**